cody.el

```diff
@@ -67,11 +68,114 @@ This is generated and cached on first use, if telemetry is enabled."
 (defcustom cody-enable-automatic-completions t
   "Non-nil to have Cody prompt with suggestions automatically.
 Completion suggestions will appear after you stop typing for a while,
-if any suggestions are available. Set this to nil if you prefer to
+if any suggestions are available.  Set this to nil if you prefer to
 use manual completion triggering with `cody-request-completion'."
   :type 'boolean
   :group 'cody)
 
+;;; Completions "API" for the rest of the lisp code:
+
+(defclass cody-completion-item ()
+  ((insertText :initarg :insertText
+               :initform nil
+               :type (or null string)
+               :documentation "The original suggested text.")
+   (range :initarg :range
+          :initform nil
+          :type (or null list)
+          :documentation "The range where the completion applies.")
+   (-displayText :initarg :displayText
+                :initform nil
+                :type (or null string)
+                :documentation "The updated text as the user types."))
+  "Represents a single alternative/suggestion in a completion response.")
+
+(cl-defmethod cody-get-display-text ((item cody-completion-item))
+  "Get the display text for the completion item ITEM."
+  (oref item -displayText))
+
+(cl-defmethod cody-set-display-text ((item cody-completion-item) value)
+  "Set the display text for the completion item ITEM to VALUE."
+  (oset item -displayText value))
```

Comment:
is it convention to add these helper functions? Direct use of oref/oset also seem readable. From reading the docs on defclass there is also :accessor which I think generates the get function at least.

Reply:
I was a bit torn on this one, as the options appear to be:

1. Use oref/oset directly -- succinct, but considered maybe fragile / exposing internals
2. Use :accessor -- this works, but then you should use setf for writes, which is stylistically maybe somewhat verbose, but 🤷
3. Create my own accessors -- not very lispy and doesn't save much text

I'll switch this to either :accessor or oref/oset.

Reply:
I switched to :accessor and you're 100% right -- it cleaned up the code a lot. Thanks!